cpp/src/community/ecg.cu

```diff
@@ -114,37 +138,34 @@ void ecg(raft::handle_t const &handle,
          vertex_t ensemble_size,
          vertex_t *clustering)
 {
+  using graph_type = GraphCSRView<vertex_t, edge_t, weight_t>;
```

Comment:
We should try and get rid of legacy classes in SG Louvain and ECG to use `graph_t`. Or the MG path could be used to support the 1 GPU case.

Reply:
There's work currently scheduled for 0.19 that will adapt all of these to use the graph primitives.  We should be able to consider getting rid of the legacy versions at that point.